src/machine/machine_atsamd21.go

```diff
@@ -1477,6 +1477,8 @@ func handleUSB(intr interrupt.Interrupt) {
 			// clear stall request
 			setEPINTENCLR(0, sam.USB_DEVICE_EPINTENCLR_STALL1)
 		}
+	} else {
+		sendZlp()
```

Comment:
This line appears to have caused the regression. What is it supposed to be doing?

Reply:
Interesting. I added to send a zlp when the other cases were not handled, which seemed to correct an issue with the USB enumeration on Windows 10, based on a comment from @sago35

Removing that line gets it to work on Linux and macOS, but does not work on Windows 10. I suppose that means we need to further investigate what Windows in sending as the initial request.

Reply:
I made a proposal for sendZlp() below and in the comments that follow.
However, I'm not sure that this idea is correct.
https://github.com/tinygo-org/tinygo/pull/884#issuecomment-594490977

The samd51 I'm trying has five USB Device EndPoint Interrupt Flags.

https://www.mouser.com/datasheet/2/268/60001507A-1130176.pdf
P.1161.

I did an additional check on atsamd51 and USB-CDC worked if you called sendZlp() on TRCPT0.
It may be better to process TRCPT0 as follows.
My environment is Windows 10 + PyPortal (samd51).

https://github.com/sago35/tinygo/commit/f80cc339e5c2f18b559a2f94f59352c05329a816.

Since sendZlp() clears the bottom 12 bits of PCKSIZE, it assumes that BYTE_COUNT = 0.
The PCKSIZE is described on P.1197.

Reply:
URB_FUNCTION_GET_MS_FEATURE_DESCRIPTOR, which probably exists only in Windows, needs to be handled well.
Maybe a careful reading of the Aruduino and circuitpython sources will solve the problem.

https://github.com/tinygo-org/tinygo/pull/884#issuecomment-594490977

Reply:
https://github.com/adafruit/ArduinoCore-samd/blob/master/cores/arduino/USB/USBCore.cpp

Reply:
This process is not in machine_atsamd51.go.
On the else side, sendZlp () may not be necessary or should not be called.

https://github.com/adafruit/ArduinoCore-samd/blob/master/cores/arduino/USB/USBCore.cpp#L951

Reply:
Please see https://github.com/tinygo-org/tinygo/pull/1013 I think you have the right idea @sago35